src/parser.rs

```diff
@@ -988,16 +988,25 @@ impl Parser {
     /// Parse a SQL CREATE statement
     pub fn parse_create(&mut self) -> Result<Statement, ParserError> {
         if self.parse_keyword(Keyword::TABLE) {
-            self.parse_create_table()
+            self.parse_create_table(false)
+        } else if self.parse_keywords(&[Keyword::OR, Keyword::REPLACE, Keyword::TABLE]) {
+            self.parse_create_table(true)
         } else if self.parse_keyword(Keyword::INDEX) {
             self.parse_create_index(false)
         } else if self.parse_keywords(&[Keyword::UNIQUE, Keyword::INDEX]) {
             self.parse_create_index(true)
         } else if self.parse_keyword(Keyword::MATERIALIZED) || self.parse_keyword(Keyword::VIEW) {
             self.prev_token();
-            self.parse_create_view()
+            self.parse_create_view(false)
+        } else if self.parse_keywords(&[Keyword::OR, Keyword::REPLACE])
+            && (self.parse_keyword(Keyword::MATERIALIZED) || self.parse_keyword(Keyword::VIEW))
```

Comment:
For `CREATE OR REPLACE MATERIALIZED VIEW` and for `CREATE OR REPLACE EXTERNAL TABLE` below please either add a test, mentioning which dialects support that or don't add support parsing that.

Reply:
:+1:

Reply:
Added tests for those options as well